libp2p/protocol_muxer/multiselect.py

```diff
@@ -19,8 +19,8 @@ class Multiselect(IMultiselectMuxer):
 
     handlers: Dict[TProtocol, StreamHandlerFn]
 
-    def __init__(self) -> None:
-        self.handlers = {}
+    def __init__(self, default_handlers: Dict[TProtocol, StreamHandlerFn] = {}) -> None:
```

Comment:
Should we initialize `default_handlers = {}` in the function body to avoid one `{}` is shared in all `Multiselect` instance?

Reply:
yes, nice catch -- i was thinking python wouldn't do this for `{}` but it ends up as a reference type so we get the clobbering behavior

Reply:
I believe [`flake8-bugbear`](https://github.com/PyCQA/flake8-bugbear) would have caught this in CI.

Reply:
good to know, i'll make an issue to add it